contracts/src/v0.8/ccip/test/pools/GHO/GhoTokenPoolEthereum.t.sol

```diff
@@ -0,0 +1,988 @@
+
+    uint256 balancePre = s_token.balanceOf(OWNER);
+    s_token.approve(address(s_ghoTokenPool), amount);
+    s_ghoTokenPool.provideLiquidity(amount);
+
+    s_ghoTokenPool.withdrawLiquidity(amount);
+
+    assertEq(s_token.balanceOf(OWNER), balancePre);
+  }
+
+  // Reverts
+
+  function test_UnauthorizedReverts() public {
+    vm.startPrank(STRANGER);
+    vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, STRANGER));
+
+    s_ghoTokenPool.withdrawLiquidity(1);
+  }
+
+  function testInsufficientLiquidityReverts() public {
+    uint256 maxUint128 = 2 ** 128 - 1;
+
+    s_token.approve(address(s_ghoTokenPool), maxUint128);
+    s_ghoTokenPool.provideLiquidity(maxUint128);
+
+    changePrank(address(s_ghoTokenPool));
+    s_token.transfer(OWNER, maxUint128);
+    changePrank(OWNER);
+
+    vm.expectRevert(LockReleaseTokenPool.InsufficientLiquidity.selector);
+    s_ghoTokenPool.withdrawLiquidity(1);
+  }
+}
+
+contract GhoTokenPoolEthereum_transferLiquidity is GhoTokenPoolEthereumSetup {
+  UpgradeableLockReleaseTokenPool internal s_oldLockReleaseTokenPool;
+
+  uint256 internal s_amount = 100_000_000e18;
+
+  error BridgeLimitExceeded(uint256 limit);
+  error InsufficientLiquidity();
+
+  function setUp() public virtual override {
+    super.setUp();
+
+    s_oldLockReleaseTokenPool = UpgradeableLockReleaseTokenPool(
+      _deployUpgradeableLockReleaseTokenPool(
+        address(s_token),
+        address(s_mockRMN),
+        address(s_sourceRouter),
+        AAVE_DAO,
+        INITIAL_BRIDGE_LIMIT,
+        PROXY_ADMIN
+      )
+    );
+    deal(address(s_token), address(s_oldLockReleaseTokenPool), s_amount);
+    changePrank(AAVE_DAO);
+    s_oldLockReleaseTokenPool.setCurrentBridgedAmount(s_amount);
+  }
+
+  function testFuzz_TransferLiquidity(uint256 amount) public {
+    amount = bound(amount, 1, s_amount);
+
+    s_oldLockReleaseTokenPool.setRebalancer(address(s_ghoTokenPool));
+
+    s_ghoTokenPool.transferLiquidity(address(s_oldLockReleaseTokenPool), amount);
+
+    assertEq(s_token.balanceOf(address(s_ghoTokenPool)), amount);
+    assertEq(s_token.balanceOf(address(s_oldLockReleaseTokenPool)), s_amount - amount);
+  }
+
+  // Reverts
+
+  function test_UnauthorizedReverts() public {
+    changePrank(STRANGER);
+    vm.expectRevert(OnlyCallableByOwner.selector);
+
+    s_ghoTokenPool.transferLiquidity(address(1), 1);
+  }
+
+  function testFuzz_RevertsTransferLiquidityExcess(uint256 amount) public {
+    uint256 existingLiquidity = s_token.balanceOf(address(s_oldLockReleaseTokenPool));
+    amount = bound(amount, existingLiquidity + 1, type(uint256).max);
+
+    s_oldLockReleaseTokenPool.setRebalancer(address(s_ghoTokenPool));
+
+    vm.expectRevert(InsufficientLiquidity.selector);
+    s_ghoTokenPool.transferLiquidity(address(s_oldLockReleaseTokenPool), amount);
+  }
+}
+
+contract GhoTokenPoolEthereum_setCurrentBridgedAmount is GhoTokenPoolEthereumSetup {
+  function setUp() public override {
```

Comment:
this is not needed right

Reply:
nope, will rm

Reply:
[5cff384](https://github.com/aave/ccip/pull/20/commits/5cff38497bb7ed8b0fd4d7aea1fe8ccf4344e22a)